src/Output/MemoryOutputBuffer.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Console\Output;
+
+use Tempest\Highlight\Themes\TerminalStyle;
+
+final class MemoryOutputBuffer implements OutputBuffer
```

Comment:
Implementation for testing where output should be written to memory, which we can use to make assertions